server/svix-server/src/v1/endpoints/endpoint/mod.rs

```diff
@@ -216,18 +216,31 @@ impl From<EndpointHeaders> for EndpointHeadersOut {
 #[serde(rename_all = "camelCase")]
 pub struct EndpointHeadersPatchIn {
     #[validate]
-    pub headers: EndpointHeaders,
+    pub headers: EndpointHeadersPatch,
 }
 
 impl ModelIn for EndpointHeadersPatchIn {
     type ActiveModel = endpoint::ActiveModel;
 
     fn update_model(self, model: &mut Self::ActiveModel) {
+        let headers: EndpointHeaders = self
+            .headers
+            .0
+            .iter()
+            .filter(|(_, v)| v.is_some())
+            .map(|(k, v)| (k.clone(), v.clone().unwrap()))
```

Comment:
Could you do a `filter_map` here instead of `.filter(...).map(...)`?

Reply:
👀 did not know that was a thing! Done.